Makefile

```diff
@@ -29,6 +29,12 @@ test_unit:
 	stat ../test-project
 	python -m doctest tests/e2e/conftest.py
 
-.PHONY: test_e2e
-test_e2e:
-	pytest -v -s --reruns=2 tests/e2e
+.PHONY: test_e2e_dev
+test_e2e_dev:
+	export TRAINING_MACHINE_TYPE=cpu-small
+	pytest -v -s --environment=dev tests/e2e
```

Comment:
Should we have `reruns=2` here as well?

Reply:
I'm not sure: the logic encoded here was to make staging tests more robust (`--reruns=2`) and to make dev tests more informative (`--reruns=0` for detecting all kinds of flakes instantly). So I propose to stick with this scheme for the first time, and later adjust values.

Reply:
I see, that makes sense now